plugins/fileservices/fileservices.cpp

```diff
@@ -2770,17 +2770,10 @@ static void checkExternalFilePath(ICodeContext *ctx,IPropertyTree *plane,const c
         dlfn.setExternal(host,path); //for backward compatibility
     dlfn.getExternalFilename(rfn);
 
-    Owned<IFile> f = createIFile(rfn);
-    if (rd && !f->exists())
-        throw makeStringExceptionV(-1,"File %s not found",path);
-
-    if (f->exists() && (f->isDirectory()==fileBool::foundYes))
-        checkExternalFileRights(ctx,dlfn.get(),rd,wr); //check if the folder can be read/writen.
-    else
-    {
-        StringBuffer scopes;
-        checkExternalFileRights(ctx,dlfn.getScopes(scopes),rd,wr);
-    }
+    //If the path is a folder, check if it can be accessed.
+    //If the path is a file, the LDAP will skip the file layer (because it is undefined in the scope access setting)
+    //and check if its scope can be accessed.
+    checkExternalFileRights(ctx,dlfn.get(),rd,wr);
```

Comment:
see comment below (related to previous comment : https://github.com/hpcc-systems/HPCC-Platform/pull/18275#discussion_r1485986966)

If this function is to be used in context where you do not know whether the user supplied path is a path to a directory, or a path to a file, you need to validate access to the parent scope e.g. "scope1::scope2" from "scope1::scope2::tail", then check what 'tail' is, and then only if it is a directory, scope check it.